parser/src/grammar.lalrpop

```diff
@@ -331,6 +332,12 @@ NoErrorPattern = {
 
         Pattern::Constructor(id, args)
     },
+
+    <id: Ident> "@" <pat: Sp<AtomicPattern>> => {
```

Comment:
Can it be `Pattern` instead of `AtomicPattern``? Otherwise `x@Some y` won't work. Or should we keep it as is and require `x@(Some y)`?

Reply:
It will be nontrivial to make this work, I think, because of grammar ambiguities. I think it could be done if we mucked around with the rules, but I'd like to do that as a separate issue.